bin/cron/hoc_student_name_cleanup

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/env ruby
+require_relative 'only_one'
+abort 'Script already running' unless only_one_running?(__FILE__)
+
+# This script deletes the name stored in pegasus.hoc_activity after 3 months.
+#
+# Background:
+# The name field is populated when the user personalizes the student certificate
+# after completing an HoC activity. This allows us to create the personalized
+# certificate when it is requested, e.g. when someone clicks on a link that the
+# student has shared to social media. For privacy reasons, we delete the name after
+# 3 months. If the certificate is requested after that point, we return the default
+# unpersonalized certificate.
+
+require File.expand_path('../../../pegasus/src/env', __FILE__)
+require 'cdo/db'
+require 'cdo/properties'
+require 'optparse'
+
+CDO.log.info("Script started at #{Time.now}")
+
+HOC_ACTIVITY = PEGASUS_DB[:hoc_activity]
+START_ID_PROPERTY_KEY = "hoc_activity.name_cleanup.start_id"
+
+# Parse options
+options = {
+  retention_days: 90,
+  retain_after_date: nil, # calculated from retention_days
+  start_id: nil,
+  end_id: nil,
+  batch_limit: 100,
+  sleep_msecs: 100,
+  dry_run: false,
+  log_sql: false,
+}
+OptionParser.new do |opts|
+  opts.banner = <<~BANNER
+    Usage: #{File.basename(__FILE__)} [options]
+
+    This script deletes names stored in the pegasus.hoc_activity table after the
+    configured retention period.
+
+    Options:
+  BANNER
+
+  opts.on('--retention=90',
+    Integer,
+    'Retention period in days. Defaults to 90 days if not specified.'
+  ) do |retention_days|
+    options[:retention_days] = retention_days
+  end
+
+  opts.on('--end-id=1000000',
+    Integer,
+    'Id of last entry to clean up (exclusive). If both this and rentention_days ' \
+    'are specified, this value takes precedence.'
+  ) do |end_id|
+    options[:end_id] = end_id
+  end
+
+  opts.on('--start-id=1',
+    Integer,
+    'Id of first entry to clean up (inclusive). If not specified, the script will ' \
+    'determine where to start based on where the last run left off.'
+  ) do |start_id|
+    options[:start_id] = start_id
+  end
+
+  opts.on('--batch-limit=100',
+    Integer,
+    'Maximum number of rows to return in each batch. Defaults to 100 if not specified.'
+  ) do |batch_limit|
+    options[:batch_limit] = batch_limit
+  end
+
+  opts.on('--sleep=100',
```

Comment:
is this to keep from locking up the db? might be nice to include what its for in this comment

Reply:
Yes, this is to reduce the performance impact on the db.  I've updated the usage text to include this, thanks!